InternalDocs/stackrefs.md

```diff
@@ -0,0 +1,79 @@
+# Stack references (`_PyStackRef`)
+
+Stack references are the interpreter's tagged representation of values on the evaluation stack.
+They carry metadata to track ownership and support optimizations such as tagged small ints.
+
+## Shape and tagging
+
+- A `_PyStackRef` is a tagged pointer-sized value (see `Include/internal/pycore_stackref.h`).
+- Tag bits distinguish three cases:
+  - `Py_TAG_REFCNT` clear - reference count lives on the pointed-to object.
+  - `Py_TAG_REFCNT` set - ownership is "borrowed" (no refcount to drop on close) or the object is immortal.
+  - `Py_INT_TAG` - tagged small integer stored directly in the stackref (no heap allocation).
+- Special constants: `PyStackRef_NULL`, `PyStackRef_ERROR`, and embedded `None`/`True`/`False`.
+
+In GIL builds, most objects carry their refcount; tagged borrowed refs skip decref on close. In free
+threading builds, the tag is also used to mark deferred refcounted objects so the GC can see them and
+to avoid refcount contention for short-lived stack values.
```

Comment:
I'm probably misreading this as it's written (it's Friday afternoon) but the following is clearer to me
```suggestion
to avoid refcount contention on commonly shared objects.
```